### DISCUSSION TITLE
Merged houses devolving

### DISCUSSION THREAD

**Discussion by catalin-duta** [Keriew/augustus#524] [Ideas]
Given the option that "all houses merge", is there any way to keep them merged in the same location, even after devolving?

I've had houses devolve into 4 1x1s that were originally:
- merged from the small tent phase (4 housing plots) - so never a 1x1
- merged some time during the evolution
- started as a single 1x1 which evolved to medium insula

I sometimes get unexpected chain reactions, in that:
- I build a workshop somewhere
- it is affecting the desirability of a nearby house which devolves
- in turn, that house has an effect on others which end up getting un-merged
- after fixing the desirability of the original location, the houses evolve back, but many times they merge back differently, leaving me with 1x1s

**Comment by crudelios** [comment]
Good idea. If 2x2's devolve to 1x1's, they should be kept merged if:

- "All houses merge" is on; or
- The terrain allows for merging.

I believe this will fix your issues.

**Reply by catalin-duta** [reply]
It definitely would.
Not sure how vanilla behaves, but, logically thinking, if I place 1 housing plot that evolves up to medium insula (turning 2x2), why would it devolve into 4 separate 1x1s?

It's not so much the devolving into 1x1s, but the fact that they evolve back on different positions that is frustrating. Especially at the edge of fountain coverage and leaving small tents as an outcome.

**Comment by crudelios** [comment]
The original behaves like augustus with "all houses merge" turned off.

The original behaviour is: depending on the terrain type (a value hidden from the user), some 1x1's will be merged to 2x2's and some won't.

When devolving, all large insulae turned into 4 1x1 medium insulae which then merged back if allowed to according to the terrain. This is usually harmless as they'll merge in the same spot, but sometimes they won't.

I just changed the code to prevent this split/almost immediate remerge. Houses will remain merged instead, **unless** the "All houses merge" option is off and the terrain type won't allow merging. In that case, large insulae will split back to 4 1x1 medium insulae.

Can you please test the new build to see if it's working as intended? I could not test it as I'm not at home.

**Reply by catalin-duta** [reply]
So far so good. I even went through a generation change where almost a third of the working population retired. Services were quite impacted, but I did not see any house splitting up. After restabilizing everything seems to be in the previous location.

**Reply by crudelios** [reply]
Good, then it's fixed 👍
